compiler/rustc_middle/src/ty/context.rs

```diff
@@ -418,6 +418,12 @@ pub struct TypeckResults<'tcx> {
     /// Stores the type, expression, span and optional scope span of all types
     /// that are live across the yield of this generator (if a generator).
     pub generator_interior_types: Vec<GeneratorInteriorTypeCause<'tcx>>,
+
+    /// We treat byte string literals as polymorphic over their type. This means
+    /// they are simultaneously `&[u8]` and `&[u8; N]`, depending on the pattern
+    /// in which they are used. This hashset records all instances where we behave
+    /// like this to allow other code to reliably process this.
+    pub polymorphic_byte_string_literals: ItemLocalSet,
```

Comment:
So this tells us which consts have a polymorphic type. How do we know which of the two types is the correct one? Also from what's below I understand that the array type is still the one assigned by default, could you mention this here?

Reply:
actually, this field should just be `treat_byte_string_as_slice`, that's the only situation where the types differ (when we get a byte string literal, which is always of `&[u8; N]` type, and we need a slice)

Reply:
Oh I see. Your latest change makes it clearer.